File: Programming101/week0/python_simple_problem_set/count_consonants.py
def count_consonants(str_):
    consonants = "bcdfghjklmnpqrstvwxzBCDFGHJKLMNPQRSTVWXZ"
    count = 0
    for ch in str_:
        if ch in consonants:
            count += 1
    return count


def main():
    print(count_consonants("Python"))
    print(count_consonants("Theistareykjarbunga"))
    print(count_consonants("grrrrgh!"))
    print(count_consonants("Github is the second best thing that happend to programmers, after the keyboard!"))
    print(count_consonants("A nice day to code!"))


if __name__ == '__main__':
    main()

File: FrontendJavaScript/week0/2-A-Set-Of-Problems/set1/countConsonants.js
"use strict";


var countConsonants = function(str) {
  var new_str = str.toLowerCase();
  var countVowels = require("./countVowels").countVowels;
  new_str = new_str.replace(/\W+/g, "");
  new_str = new_str.replace(/\s/g, "");
  return new_str.length - countVowels(new_str);
};

console.log(countConsonants("JavaScript") === 7);
console.log(countConsonants("Theistareykjarbunga") === 11);
console.log(countConsonants("grrrrgh!") === 7);
console.log(countConsonants("Github is the second best thing that happend to programmers, after the keyboard!") === 44);
console.log(countConsonants("A nice day to code!") === 6);
